suite-native/discovery/src/utils.ts

```diff
@@ -1,38 +1,37 @@
 import { A, F } from '@mobily/ts-belt';
 
-import { NetworkCompatible, AccountType, NetworkSymbol } from '@suite-common/wallet-config';
+import { AccountType, Network, NetworkSymbol } from '@suite-common/wallet-config';
 import { Account } from '@suite-common/wallet-types';
 
-const NORMAL_ACCOUNT_TYPE = 'normal';
-
-// network uses undefined for normal type, but account uses 'normal'
-const normalizedAccountType = (accountType?: AccountType) => accountType ?? NORMAL_ACCOUNT_TYPE;
-
 export const getNetworksWithUnfinishedDiscovery = (
-    enabledNetworks: readonly NetworkCompatible[],
+    enabledNetworks: readonly Network[],
     accounts: Account[],
     accountsLimit: number,
 ) =>
     enabledNetworks.filter(network => {
-        const networkAccountsOfType = accounts.filter(
-            account =>
-                account.symbol === network.symbol &&
-                account.accountType === normalizedAccountType(network.accountType),
-        );
-
         // if there is no account for this network -> we should have at least one account even if added via Add Coin
         // or if there is at least one visible account of this type, we should have at least one hidden account so adding funds outside of this app is detected and account shown
-        return (
-            A.isEmpty(networkAccountsOfType) ||
-            (networkAccountsOfType.length < accountsLimit &&
-                networkAccountsOfType.every(account => account.visible))
-        );
+        // the whole network is considered undiscovered if this is found for either 'normal' account, or any of the other accountTypes
+        const isUndiscoveredAccountType = (accountType: AccountType) => {
+            const networkAccountsOfType = accounts.filter(
+                account => account.symbol === network.symbol && account.accountType === accountType,
+            );
+
+            return (
+                A.isEmpty(networkAccountsOfType) ||
+                (networkAccountsOfType.length < accountsLimit &&
+                    networkAccountsOfType.every(account => account.visible))
+            );
+        };
+
+        if (isUndiscoveredAccountType('normal')) return true;
```

Comment:
I can create a helper for it - a function `Network => NetworkAccount[]`, but with 1) `'normal'` account constructed from Network properties 2) missing properties for alternative accounts will be backfilled from `Network`.

So I will never again have to do stuff like `account.features ?? network.features` etc.

It will be helpful also for `suite` :slightly_smiling_face:

Reply:
pretty, please :) 🙏

Reply:
Created in https://github.com/trezor/trezor-suite/pull/14281/commits/8f0270267d714f747542facdee446861ef20ed69, used in https://github.com/trezor/trezor-suite/commit/b92c24e0f3a100b1c17fa3ff3856b5b2c9240860